### PR TITLE
Add basic UI

### PR DESCRIPTION
**Pull request by Kaiserabbas**
# Console APP Basic UI
- Created a app.rb file that will serve as your console app entry-point. It should have methods that do the following:
  - List all books.
  - List all people.
  - Created a person (teacher or student, not a plain Person).
  - Created a book.
  - Created a rental.
  - Listed all rentals for a given person id.
- In main.rb defined the entry point, this is a method called main that is invoked at the end of file. This method does the following:
  - Presented the user with a list of options to perform.
  - Let users choose an option.
  - Asked for parameters for the option.
  - Had a way to quit the app.

### CODE REVIEW

File: src/app.rb [hunk 1]
```diff
@@ -0,0 +1,168 @@
...
+    # Add the created objects to your data
+    @people.push(student1, student2, teacher1)
+    @books.push(book1, book2)
+    @rentals.push(rental1, rental2)
+  end
+
+  # List of books...
+  def list_books
+    puts(@books.map { |book| "Title: \"#{book.title}\", Author \"#{book.author}\"" })
+  end
```
Comment (Efiamotu-1) [inline src/app.rb:21]: - [ ] When i click on option 1 to list all books at the start of the application, i can see that there are already created books as in the image below

- [ ] After adding two books, i can also see the newly added books with the already added books as shown in the image below, however your app should not function this way. only books added at the start of the app should be dipslayed as done in this [reference video](https://www.youtube.com/watch?v=vkkgrhD6aXQ)

File: src/app.rb [hunk 2]
```diff
@@ -0,0 +1,168 @@
...
+
+  # List of people...
+  def list_people
+    puts(@people.map { |person| "[#{person.class}] Name: #{person.name}, ID: #{person.id}, Age: #{person.age}" })
+  end
```
Comment (Efiamotu-1) [inline src/app.rb:26]: - [ ] Please only display the added users once the app is run as shown in the [reference video](https://www.youtube.com/watch?v=vkkgrhD6aXQ). you are not required to have already created teacher or student :+1: :sparkles:

File: src/app.rb [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+    @people[person_index] if (0...@people.length).cover?(person_index)
+  end
+
+  # List all rentals for a given person ID
+  def list_rentals_for_person
+    print 'Enter the ID of the person: '
+    person_id = gets.chomp.to_i
+
+    person = @people.find { |p| p.id == person_id }
+
```
Comment (Efiamotu-1) [inline src/app.rb:129]: - [ ] There are already created rentals as well for your application which is not supposed to be so, rentals should only be added after creating new users and books. :-1: :sparkles:

**Review by Efiamotu-1 — CHANGES_REQUESTED** [review]
Hi @Kaiserabbas , 

 

Good job so far! 👏. Your Code is Awesome! 💪

There are some **few** issues 🐛 that you still need to work 🔧 on though to go to the next project ⏭️ but you are almost there! 🏃
## Don't Worry 🙏

It's ok to have bugs in your code 🐛. Please, see this code review as a **chance** for you to **learn** practical coding skills and learn **new tips** that can improve your code quality 💯 and help you **get a better ** 💼.

_I am going to guide you on how to fix them down below. 🤓_

## Highlights :dart: 
- ✔️ No linter Errors
- ✔️ Descriptive README
- ✔️ Descriptive PR title and description
- The code is top-notch! 💯 and has a huge potential.

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._


## Important Notes 📝

- Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
- Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

> **Note:** Please, don't Merge This Pull Request yet. Only Merge when it is Approved ✅.


## Thank you for your time 🙏 and call me if you need help or clarification ☎️

- This is My **GitHub**: [@Efiamotu-1](https://github.com/Efiamotu-1/)
- This is My **Twitter**: [@EFYAMOTU](https://twitter.com/EFYAMOTU)
- This is My **linkedIn**.[Habeeb Efiamotu Musa](https://www.linkedin.com/in/musa-habeeb/)
- This is My **Slack**: [Habeeb Efiamotu Musa](https://slack.com)

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: src/app.rb (hunk 3 above)
Comment (Kaiserabbas) [reply src/app.rb:129]: [Update the code by removing initial data supplied to the app](https://github.com/Kaiserabbas/school_library/pull/4/commits/5fb09cf98f3d48509812bb27f0ce5dbc12be1d40)

File: src/app.rb (hunk 1 above)
Comment (Kaiserabbas) [reply src/app.rb:21]: [Update the code by removing initial data supplied to the app](https://github.com/Kaiserabbas/school_library/pull/4/commits/5fb09cf98f3d48509812bb27f0ce5dbc12be1d40)

File: src/app.rb (hunk 2 above)
Comment (Kaiserabbas) [reply src/app.rb:26]: [Update the code by removing initial data supplied to the app](https://github.com/Kaiserabbas/school_library/pull/4/commits/5fb09cf98f3d48509812bb27f0ce5dbc12be1d40)

**Review by KayLemba — APPROVED** [review]
[[# STATUS: APPROVED :green_circle: :clap: 

Hello @Kaiserabbas   , 
Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit:  . 
Congratulations! 🎉  Keep up the Awesome work 🥂  Keep Rocking 🥇 

HIGHLIGHTS:

- [x]  Your repo is professional  😎 
- [x] Descriptive P.R 💯 
- [x] Professional README file 
- [x] App running as intended. 💯 🥇 
- [x] Good Job implementing the changes requested by previous code reviewer





 

 










 








Now you can merge this branch and move on :1st_place_medal: 
Happy Coding cheers 🥇 💯 



well-1

Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better.



Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._]]

**Pull request merged**